### PR TITLE
Update REST namespace separator to use unit separator char

### PR DESCRIPTION
**Pull request by danielcweeks**
The null byte character is not generally considered safe for use in REST paths even when url encoded due to various forms of null byte injection attacks.  This will cause problems with many frameworks that detect and reject requests using this character in the path.

The unit separator character (`0x1F`) is a safer and more meaningful alternative.

### CODE REVIEW

**Review by rdblue — CHANGES_REQUESTED** [review]
This also needs to update the joiner and splitter in `RESTUtil` and the associated tests in:
* `TestResourcePaths`
* `TestCreateNamespaceRequest`
* `TestRESTUtil`
* `TestCreateNamespaceResponse`
* `TestGetNamespaceResponse`
* `TestListTablesResponse`

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -197,7 +197,7 @@ paths:
           description:
             An optional namespace, underneath which to list namespaces.
             If not provided or empty, all top-level namespaces should be listed.
-            If parent is a multipart namespace, the parts must be separated by the null byte.
+            If parent is a multipart namespace, the parts must be separated by the unit separator (`0x1F`) byte.
```
[the remark this answers is not in the document]
Comment (rdblue) [reply open-api/rest-catalog-open-api.yaml:200]: @danielcweeks, there are more instances of `%00` in this file.

**Review by rdblue — APPROVED** [review]

**Review by nastra — APPROVED** [review]
LGTM, just wondering whether we should still support the old format for backward compatibility. It's probably ok to just switch to the new formt before doing a 1.0 release

**Comment by kbendick** [comment]
Belated +1. We'd discussed this offline. Thanks @danielcweeks!

**Pull request merged**